.github/workflows/benchmark_base.yml

```diff
@@ -0,0 +1,30 @@
+on:
+  push:
+    branches: main
+
+jobs:
+  benchmark_base_branch:
+    name: Continuous Benchmarking with Bencher
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9' 
+      - uses: actions/cache@v2
+        with:
+          path: .venv
+          key: poetry-${{ hashFiles('poetry.lock') }}
+      - uses: bencherdev/bencher@main
+      - name: Track base branch benchmarks with Bencher
+        run: |
+          . scripts/ci_install_deps
+          bencher run \
+          --project pyquil \
+          --token '${{ secrets.BENCHER_API_TOKEN }}' \
+          --branch master \
+          --testbed ci-runner-linux \
+          --err \
+          -- file results.json \ 
+          poetry run pytest --benchmark-json results.json test/benchmarks 
```

Comment:
Why not add a Makefile recipe for this? Is `bencher` only really useful in CI?

Reply:
`--testbed ci-runner-linux` is a parameter specific to the CI runner, it would be a mistake to publish results to it from a local machine.

`bencher` could be used locally if a user wanted to setup their own testbed, but my hope is that CI generates all the useful data a user would want.